internal/backoff/backoff_test.go

```diff
@@ -472,74 +239,396 @@ func Test_backoff_Close(t *testing.T) {
+		}
+		if !reflect.DeepEqual(gotRes, w.wantRes) {
+			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", gotRes, w.wantRes)
+		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       fields: fields {
-		           wg: sync.WaitGroup{},
-		           backoffFactor: 0,
-		           initialDuration: 0,
-		           jittedInitialDuration: 0,
-		           jitterLimit: 0,
-		           durationLimit: 0,
-		           maxDuration: 0,
-		           maxRetryCount: 0,
-		           backoffTimeLimit: nil,
-		           errLog: false,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           fields: fields {
-		           wg: sync.WaitGroup{},
-		           backoffFactor: 0,
-		           initialDuration: 0,
-		           jittedInitialDuration: 0,
-		           jitterLimit: 0,
-		           durationLimit: 0,
-		           maxDuration: 0,
-		           maxRetryCount: 0,
-		           backoffTimeLimit: nil,
-		           errLog: false,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
+			err := errors.New("error is occurred")
+			f := func(context.Context) (interface{}, bool, error) {
+				return nil, false, err
+			}
+			return test{
+				name: "return nil response and nil error when function returns (nil, false, error)",
+				args: args{
+					ctx: ctx,
+					f:   f,
+				},
+				want: want{
+					err: err,
+				},
+				checkFunc: defaultCheckFunc,
+				afterFunc: func(args) {
+					cancel()
+				},
+			}
+		}(),
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
+			f := func(context.Context) (interface{}, bool, error) {
+				return nil, true, nil
+			}
+			return test{
+				name: "return nil response and nil error when function returns (nil, true, nil)",
+				args: args{
+					ctx: ctx,
+					f:   f,
+				},
+				want:      want{},
+				checkFunc: defaultCheckFunc,
+				afterFunc: func(args) {
+					cancel()
+				},
+			}
+		}(),
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
+			err := errors.New("erros is occurred")
+			f := func(context.Context) (interface{}, bool, error) {
+				return nil, false, err
+			}
+			return test{
+				name: "return response and error when function return (nil, true, error) and maxRetryCount = 0",
+				args: args{
+					ctx: ctx,
+					f:   f,
+				},
+				fields: fields{
+					wg:                    sync.WaitGroup{},
+					backoffFactor:         0,
+					initialDuration:       0,
+					jittedInitialDuration: 0,
+					jitterLimit:           0,
+					durationLimit:         0,
+					maxDuration:           0,
+					maxRetryCount:         0,
+					backoffTimeLimit:      0,
```

Comment:
Can we remove this code, because these values are default values?:thinking:

```suggestion
```

Reply:
@hlts2 
we can remove it.
which one is the best for readable code?

Reply:
it is difficult 😢  I have no answer ...
For the time being, I think it's okay as your implementation! I will close the comment sorry.
If I find the answer, I'll let you know again.

Reply:
I think if the value is meaningful to the test case , we can/should write the value here.
what do you think about it? @hlts2 @vankichi

Reply:
@hlts2 
At once, I do not change any more about these codes.
plz, review again.